File: HackerRank Solutions/Mathematics/Constructing a Number.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int t;
    cin >> t;
    for(int i= 0;i<t;i++){
        long int n, sum=0;
        cin >> n;
        vector <int> a(n);
        for(int j=0;j<n;j++){
           cin >> a[j];
            sum+=a[j];
        }
        if(sum%3==0){
            cout<<"Yes"<<endl;
        } else{
            cout<<"No"<<endl;
        }
    }
    return 0;
}

File: HackerRank Solutions/Mathematics/Constructing a Number.py
#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the canConstruct function below.
def canConstruct(a):
    # Return "Yes" or "No" denoting whether you can construct the required number.
    sum = 0
    for i in range(len(a)):
        sum = sum+a[i]
    if(sum%3 == 0):
        return ("Yes")
    return ("No")
if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    t = int(input())

    for t_itr in range(t):
        n = int(input())

        a = list(map(int, input().rstrip().split()))

        result = canConstruct(a)

        fptr.write(result + '\n')

    fptr.close()
